sdk/core/System.ClientModel/README.md

```diff
@@ -23,71 +22,290 @@ dotnet add package System.ClientModel
 
-- Configuring service clients (`ClientPipelineOptions`).
-- Accessing HTTP response details (`ClientResult`, `ClientResult<T>`).
-- Exceptions for reporting errors from service requests in a consistent fashion (`ClientResultException`).
-- Customizing requests (`RequestOptions`).
-- Providing APIs to read and write models in different formats (`ModelReaderWriter`).
+- Results that provide access to the service response and the HTTP response details (`ClientResult<T>`, `ClientResult`).
+- Exceptions that result from failed requests (`ClientResultException`).
+- Options used to configure the service client pipeline (`ClientPipelineOptions`).
+- Options used to customize HTTP requests (`RequestOptions`).
+- Content sent in an HTTP request body (`BinaryContent`).
+
+Below, you will find sections explaining these shared concepts in more detail.
 
 ## Examples
 
-### Send a message using the MessagePipeline
+### Send a message using ClientPipeline
+
+`System.ClientModel`-based clients, or **service clients**, use the `ClientPipeline` type to send and receive HTTP messages. The following sample shows a minimal example of what a service client implementation might look like.
+
+```C# Snippet:ReadmeSampleClient
+public class SampleClient
+{
+    private readonly Uri _endpoint;
+    private readonly ApiKeyCredential _credential;
+    private readonly ClientPipeline _pipeline;
+
+    // Constructor takes service endpoint, credential used to authenticate
+    // with the service, and options for configuring the client pipeline.
+    public SampleClient(Uri endpoint, ApiKeyCredential credential, SampleClientOptions? options = default)
+    {
+        // Default options are used if none are passed by the client's user.
+        options ??= new SampleClientOptions();
+
+        _endpoint = endpoint;
+        _credential = credential;
+
+        // Authentication policy instance is created from the user-provided
+        // credential and service authentication scheme.
+        ApiKeyAuthenticationPolicy authenticationPolicy = ApiKeyAuthenticationPolicy.CreateBearerAuthorizationPolicy(credential);
+
+        // Pipeline is created from user-provided options and policies
+        // specific to the service client implementation.
+        _pipeline = ClientPipeline.Create(options,
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: new PipelinePolicy[] { authenticationPolicy },
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+    }
+
+    public ClientResult<SampleResource> UpdateResource(SampleResource resource)
+    {
+        // Create a message that can be sent via the client pipeline.
+        PipelineMessage message = _pipeline.CreateMessage();
+
+        // Set a classifier that will determine whether the response is an
+        // error response based on the response status code.
+        message.ResponseClassifier
```

Comment:
We have a default classifier, don't we? If yes, maybe we can remove this line for the basic sample.

Reply:
Yep, simple to do that.